koku/sources/kafka_listener.py

```diff
@@ -501,33 +502,56 @@ def execute_koku_provider_op(msg, cost_management_type_id):
         msg (Asyncio msg): Dictionary messages containing operation,
                                        provider and offset.
             example: {'operation': 'create', 'provider': SourcesModelObj, 'offset': 3}
-        cost_management_type_id (Integer): Cost Management Type Identifier
 
     Returns:
         None
 
     """
     provider = msg.get("provider")
     operation = msg.get("operation")
+    account_coordinator = SourcesProviderCoordinator(provider.source_id, provider.auth_header)
+    sources_client = SourcesHTTPClient(provider.auth_header, provider.source_id)
 
     try:
         if operation == "create":
-            task = create_or_update_provider.delay(provider.source_id)
-            LOG.info(f"Creating Koku Provider for Source ID: {str(provider.source_id)} in task: {task.id}")
+            LOG.info(f"Creating Koku Provider for Source ID: {str(provider.source_id)}")
+            instance = account_coordinator.create_account(
+                provider.name,
+                provider.source_type,
+                provider.authentication,
+                provider.billing_source,
+                provider.source_uuid,
+            )
+            LOG.info(f"Creating provider {instance.uuid} for Source ID: {provider.source_id}")
         elif operation == "update":
-            task = create_or_update_provider.delay(provider.source_id)
-            LOG.info(f"Updating Koku Provider for Source ID: {str(provider.source_id)} in task: {task.id}")
+            instance = account_coordinator.update_account(
+                provider.koku_uuid,
+                provider.name,
+                provider.source_type,
+                provider.authentication,
+                provider.billing_source,
+            )
+            LOG.info(f"Updating provider {instance.uuid} for Source ID: {provider.source_id}")
         elif operation == "destroy":
-            task = delete_source_and_provider.delay(provider.source_id, provider.source_uuid, provider.auth_header)
-            LOG.info(f"Deleting Koku Provider/Source for Source ID: {str(provider.source_id)} in task: {task.id}")
+            account_coordinator.destroy_account(provider.koku_uuid)
+            LOG.info(f"Destroying provider {provider.koku_uuid} for Source ID: {provider.source_id}")
         else:
             LOG.error(f"unknown operation: {operation}")
+        sources_client.set_source_status(None)
 
+    except SourcesProviderCoordinatorError as account_error:
+        raise SourcesIntegrationError("Koku provider error: ", str(account_error))
+    except ValidationError as account_error:
+        err_msg = (
+            f"Unable to {operation} provider for Source ID: {str(provider.source_id)}. Reason: {str(account_error)}"
+        )
+        LOG.error(err_msg)
+        sources_client.set_source_status(account_error)
     except RabbitOperationalError:
```

Comment:
Do we need this error anymore? We aren't creating a celery task, right?

Reply:
yup, think this can go 👍

Reply:
Don't need the error. I'll remove it.